src/Core/Util/ArrayFinder.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+/**
+ * ArrayFinder allows to modify an array content using selectors such as $arrayFinder->get('property_a.property_3.4');
+ *
+ * This class replaces https://github.com/Shudrum/ArrayFinder/blob/master/ArrayFinder.php that
+ * was used in previous PrestaShop versions.
+ */
+class ArrayFinder implements \ArrayAccess, \Countable
+{
+    /**
+     * @var array
+     */
+    private $array;
+
+    /**
+     * @var PropertyAccessor
+     */
+    private $propertyAccessor;
+
+    /**
+     * @param array $content the array to be searched and manager by ArrayFinder
+     */
+    public function __construct(array $content = [])
+    {
+        $this->array = $content;
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        return count($this->array);
+    }
+
+    /**
+     * @param string|int|null $path selector to find the desired value. if empty, will return full array
+     * @param mixed|null $default default value to be returned if selector matches nothing
+     *
+     * @return mixed|null
+     *
+     * Examples of use:
+     * $arrayFinder->get('a');
+     * $arrayFinder->get(4);
+     * $arrayFinder[4];
+     */
+    public function get($path = null, $default = null)
```

Comment:
```suggestion
    public function get(?string $path = null, $default = null)
```

Reply:
The consequence of this change is that we cannot do
```
$finder = new ArrayFinder($anArray);
$finder->get(28);
```

but if you're OK with that, then I have implemented this change in https://github.com/PrestaShop/PrestaShop/pull/25611/commits/e3f206033e75feda1ae4cd721084e5e5fcadf09c